scaleway/resource_mnq_queue.go

```diff
@@ -0,0 +1,544 @@
+				Required:    true,
+				Sensitive:   true,
+				Description: "Line jump separated key and seed",
+			},
+		},
+	}
+}
+
+func resourceScalewayMNQQueueCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api, region, err := newMNQAPI(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	namespaceRegion, namespaceID, err := parseRegionalID(d.Get("namespace_id").(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if region != namespaceRegion {
+		return diag.Errorf("region of the queue (%s) and the namespace (%s) must match", region, namespaceRegion)
+	}
+
+	namespace, err := api.GetNamespace(&mnq.GetNamespaceRequest{
+		Region:      namespaceRegion,
+		NamespaceID: namespaceID,
+	}, scw.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	switch namespace.Protocol {
+	case mnq.NamespaceProtocolSqsSns:
+		return resourceScalewayMNQQueueCreateSQS(ctx, d, meta)
+	case mnq.NamespaceProtocolNats:
+		return resourceScalewayMNQQueueCreateNATS(ctx, d, meta)
+	default:
+		return diag.Errorf("unknown protocol %s", namespace.Protocol)
+	}
+}
+
+func resourceScalewayMNQQueueCreateSQS(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, _, err := SQSClientWithRegion(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	isFifo := d.Get("fifo_queue").(bool)
+	name := resourceMNQQueueName(d.Get("name"), d.Get("name_prefix"), true, isFifo)
+
+	attributes, err := sqsResourceDataToAttributes(d, resourceScalewayMNQQueue().Schema)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[DEBUG] SQS Queue attributes CREATE: %+v", attributes)
+
+	input := &sqs.CreateQueueInput{
+		QueueName:  aws.String(name),
+		Attributes: attributes,
+	}
+
+	_, err = retryWhenAWSErrCodeEquals(ctx, []string{sqs.ErrCodeQueueDeletedRecently}, &RetryWhenConfig[*sqs.CreateQueueOutput]{
+		Timeout:  d.Timeout(schema.TimeoutCreate),
+		Interval: defaultMNQQueueRetryInterval,
+		Function: func() (*sqs.CreateQueueOutput, error) {
+			return client.CreateQueueWithContext(ctx, input)
+		},
+	})
+	if err != nil {
+		return diag.Errorf("failed to create SQS Queue: %s", err)
+	}
+
+	namespaceRegion, namespaceID, err := parseRegionalID(d.Get("namespace_id").(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(composeMNQQueueID(namespaceRegion, namespaceID, *input.QueueName))
+
+	return resourceScalewayMNQQueueReadSQS(ctx, d, meta)
+}
+
+func resourceScalewayMNQQueueCreateNATS(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, _, err := NATSClientWithRegion(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	name := resourceMNQQueueName(d.Get("name"), d.Get("name_prefix"), false, false)
+	maxAge := d.Get("message_max_age").(int)
+	maxSize := d.Get("message_max_size").(int)
+
+	var retention nats.RetentionPolicy
+	if d.Get("fifo_queue").(bool) {
+		retention = nats.InterestPolicy
```

Comment:
Is this what we want for a fifo ? If multiple workers are waiting for a message, it will be read by all workers.
Also the message will be deleted if there is no listeners.
Look at https://docs.nats.io/nats-concepts/jetstream/streams#retentionpolicy